### PR TITLE
feat(ui): Friendlier 403 err msg for AsyncComponent

### PR DESCRIPTION
**Pull request by billyvg**
Previously would always show generic RouteError

This will reduce the number of errors reported by AsyncComponent (e.g. JAVASCRIPT-2W4)

### CODE REVIEW

File: src/sentry/static/sentry/app/components/asyncComponent.jsx [hunk 1]
```diff
@@ -110,7 +113,7 @@ class AsyncComponent extends React.Component {
               },
               remainingRequests: prevState.remainingRequests - 1,
               loading: prevState.remainingRequests > 1,
-              error: !!error,
+              error: prevState.error || !!error,
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/asyncComponent.jsx]: do we want to keep `error` at all or switch to only using `errors`? or would that complicate all the component logic?

**Review by MaxBittker — APPROVED** [review]
very nice. partly i worry that we could ship a bug to  an endpoint that sends erroneous 403s and we wouldn't find out, but i guess that it's better to reduce some of the noise here and find another solution if we think that's a real problem

File: src/sentry/static/sentry/app/components/asyncComponent.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/asyncComponent.jsx]: I think it's probably ok to have the broader/simpler `error` state here and use `errors` if we need more fine grained error details.

**Comment by billyvg** [comment]
I think that's a legit concern, that's solved pretty simply by capturing 403s into sentry

**Pull request merged**